Support using system QuaZIP on Qt5 and output build messages for system libgit2 and QuaZIP

cjmayo opened the pull request:
* Currently Qt5 builds can't use system QuaZIP because it is using quazip5 locations.
* Add build-time messages to inform about the use of system libraries
* Don't output three copies of the build-time messages

A bit more info in the commit messages.

This is an updated version of the work in #3254 and a copy of #3468.

Happy to squash commits in this PR or split into separate PRs if preferred.

cjmayo committed:
Only output messages once during Makefile generation

Project MESSAGE: using installed libgit2 library
Checking for boost... yes
Project MESSAGE: using installed Boost library
Project MESSAGE: using installed quazip library
Project MESSAGE: libs -lquazip5
Project MESSAGE: using installed libgit2 library
Project MESSAGE: using installed Boost library
Project MESSAGE: using installed quazip library
Project MESSAGE: libs -lquazip5
Project MESSAGE: using installed libgit2 library
Project MESSAGE: using installed Boost library
Project MESSAGE: using installed quazip library
Project MESSAGE: libs -lquazip5

"Note: By default, messages are written out for each Makefile generated
by qmake for a given project"
http://doc.qt.io/qt-5/qmake-test-function-reference.html#message-string

cjmayo committed:
Add message to show system QuaZIP is being used

cjmayo committed:
Support system QuaZIP on Qt 5

QuaZIP 0.7.2 added dual Qt 4/5 support by installing for Qt 5:
    /usr/include/quazip5/quazip.h
    /usr/include/quazip5/quazipfile.h
    /usr/lib64/libquazip5.so

https://github.com/stachenov/quazip/commit/cc2e8ff9bc45679c3312592cce39ccaf62c66a00

madpipeline reviewed (approved):
Looks ok to me. @KjellMorgenstern you ok if we merge this?

ovidiub13 commented:
@cjmayo on what system and with what compiler are you building this?

cjmayo commented:
I'm using Gentoo Linux with GCC 8.3.0. QuaZIP 0.7.6.

KjellMorgenstern commented:
I see two issues with this PR.
1. Linking with QuaZIP works fine, why risk breaking things?
2. Linking against a system QuaZIP is not tested in CI.

I think the preferred way to fix this should be to remove the quazip code from Fritzing entirely, and adapt the CI accordingly. This way we would only have to maintain the dependency, as with every other library, not the library itself. Opinions?

KjellMorgenstern reviewed (changes requested):
I think it is better to remove the copy of quazip from Fritzing, and properly link against the library instead.

cjmayo commented:
Removing bundled QuaZIP sounds good to me.

cjmayo commented:
Updated onto latest develop. I don't use or know much about static linking but hopefully this could be the basis for still allowing dynamic linking to a system library and modified for static linking to a Fritizing copy.
